src/Syntax/TST/Types.hs

```diff
@@ -264,22 +285,26 @@ instance Zonk (Typ pol) where
   zonk RecRep bisubst ty@(TyRecVar _ NegRep _ tv) = case M.lookup tv (bisubst_map bisubst) of
     Nothing -> ty
     Just (_,tyNeg) -> tyNeg
-  zonk vt bisubst (TyData loc rep xtors) =
-     TyData loc rep (zonk vt bisubst <$> xtors)
-  zonk vt bisubst (TyCodata loc rep xtors) =
-     TyCodata loc rep (zonk vt bisubst <$> xtors)
-  zonk vt bisubst (TyDataRefined loc rep tn xtors) =
-     TyDataRefined loc rep tn (zonk vt bisubst <$> xtors)
-  zonk vt bisubst (TyCodataRefined loc rep tn xtors) =
-     TyCodataRefined loc rep tn (zonk vt bisubst <$> xtors)
-  zonk vt bisubst (TyNominal loc rep kind tn args) =
-     TyNominal loc rep kind tn (zonk vt bisubst <$> args)
+  zonk vt bisubst (TyData loc rep mk xtors) =
+     TyData loc rep mk (zonk vt bisubst <$> xtors)
+  zonk vt bisubst (TyCodata loc rep mk xtors) =
+     TyCodata loc rep mk (zonk vt bisubst <$> xtors)
+  zonk vt bisubst (TyDataRefined loc rep mk tn xtors) =
+     TyDataRefined loc rep mk tn (zonk vt bisubst <$> xtors)
+  zonk vt bisubst (TyCodataRefined loc rep mk tn xtors) =
+     TyCodataRefined loc rep mk tn (zonk vt bisubst <$> xtors)
+  zonk UniRep bisubst (TyNominal loc rep knd tn args) = 
+    TyNominal loc rep (zonkKind bisubst knd) tn (zonk UniRep bisubst <$> args)
```

Comment:
is the use of `zonkKind` here already required here?
If not, I would probably leave it out and deal with it when the constraint generation for kinds is complete and we can see that it is actually needed.
I'm mainly saying this b/c we will need to really look into the correct zonking in the end, so doing it partially now already might make  things more confusing later.

Reply:
Since  kinds aren't actually checked yet, this doesn't make a difference, but I removed it because it may cause confusion later.
This is a leftover from the old branch I forgot toe remove